test/e2e/infinispan/scaling_test.go

```diff
@@ -116,3 +121,126 @@ func TestScaling(t *testing.T) {
+
+	tutils.ExpectNoError(
+		testKube.UpdateInfinispan(ispn, func() {
+			ispn.Spec.Replicas = 0
+		}),
+	)
+	testKube.WaitForInfinispanState(ispn.Name, ispn.Namespace, func(i *ispnv1.Infinispan) bool {
+		c := i.GetCondition(ispnv1.ConditionStopping)
+		return c.Status == metav1.ConditionFalse && strings.Contains(c.Message, "unable to proceed with GracefulShutdown as the cluster health is 'DEGRADED'")
+	})
+	cacheHelper.Available(true)
+	testKube.WaitForInfinispanPods(0, tutils.SinglePodTimeout, ispn.Name, tutils.Namespace)
+	testKube.WaitForInfinispanCondition(ispn.Name, ispn.Namespace, ispnv1.ConditionGracefulShutdown)
+}
+
+func TestGracefulShutdownBlockedOnSplitBrain(t *testing.T) {
+	defer testKube.CleanNamespaceAndLogOnPanic(t, tutils.Namespace)
+
+	replicas := 2
+	ispn := tutils.DefaultSpec(t, testKube, func(i *ispnv1.Infinispan) {
+		i.Spec.Replicas = int32(replicas)
+	})
+	testKube.CreateInfinispan(ispn, tutils.Namespace)
+	testKube.WaitForInfinispanPods(replicas, tutils.SinglePodTimeout, ispn.Name, tutils.Namespace)
+	ispn = testKube.WaitForInfinispanCondition(ispn.Name, ispn.Namespace, ispnv1.ConditionWellFormed)
+
+	networkPolicy := &networkingv1.NetworkPolicy{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "networking.k8s.io/v1",
+			Kind:       "NetworkPolicy",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "deny-all-but-user",
+			Namespace: tutils.Namespace,
+		},
+		Spec: networkingv1.NetworkPolicySpec{
+			PodSelector: metav1.LabelSelector{},
+			PolicyTypes: []networkingv1.PolicyType{
+				networkingv1.PolicyTypeEgress,
+				networkingv1.PolicyTypeIngress,
+			},
+			Egress: []networkingv1.NetworkPolicyEgressRule{
+				{
+					Ports: []networkingv1.NetworkPolicyPort{
+						{
+							Port: &intstr.IntOrString{
+								Type:   intstr.Int,
+								IntVal: 11222,
+							},
+						},
+					},
+				},
+			},
+			Ingress: []networkingv1.NetworkPolicyIngressRule{
+				{
+					Ports: []networkingv1.NetworkPolicyPort{
+						{
+							Port: &intstr.IntOrString{
+								Type:   intstr.Int,
+								IntVal: 11222,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	testKube.CreateNetworkPolicy(networkPolicy)
+	defer testKube.DeleteNetworkPolicy(networkPolicy)
+
+	_client := tutils.HTTPClientForClusterWithVersionManager(ispn, testKube, tutils.VersionManager())
+	operand := tutils.Operand(ispn.Spec.Version, tutils.VersionManager())
+
+	// TODO: Are we missing a general wait function that doesn't take into account k8s resources
+	// It should take about two minutes for the newtork policy to take effect but Operator won't notice
+	testKube.WaitForInfinispanState(ispn.Name, ispn.Namespace, func(i *ispnv1.Infinispan) bool {
+		containerInfo, err := client.New(operand, _client).Container().Info()
+
+		// The request fails the first time it's executed after NetworkPolicy is created
+		if err != nil {
+			tutils.Log().Warn(err.Error())
+			return false
+		}
+
+		return containerInfo.ClusterSize != ispn.Spec.Replicas
```

Comment:
CI is failing on this check. I'm guessing because NetworkPolicy is not playing nicely with `kind`. Did you have any luck on your local machine?